lib/rules/no-unused-expressions.js

```diff
@@ -211,7 +161,8 @@ module.exports = {
 			ExpressionStatement(node) {
 				if (
 					Checker.isDisallowed(node.expression) &&
-					!isDirective(node)
+					!ignoreDirectives &&
+					!astUtils.isDirective(node)
 				) {
 					context.report({ node, messageId: "unusedExpression" });
 				}
```

Comment:
This way, `"ignoreDirectives": true` would ignore everything, not just directives?

```js
/* eslint no-unused-expressions: ["error", { "ignoreDirectives": true }] */

foo; // false negative

```

Reply:
Let's move discussion to https://github.com/eslint/eslint/pull/19645 only.